src/include/migraphx/permutation.hpp

```diff
@@ -56,12 +56,12 @@ inline std::vector<int64_t> sort_permutation(const Vector& data, Op op)
 }
 
 /*!
- * Returns the permutation needed to apply to the shape to undo the current permutation
+ * Returns the inverse permutation that could be applied to reverse the inputted permutation
  */
 std::vector<int64_t> invert_permutation(const std::vector<int64_t>& permutation);
 
 /*!
- * Finds the permutation most likely from a transpose operator that has been applied to the shape.
+ * Finds the permutation that would put the strides in descending order
```

Comment:
This seems backwards. The permutation is from the transpose(or reorder_shape) when the strides were in descending order. The permutation cannot be used to put the strides in descending order, you need `invert_permutation` for that.

Also, it would be better to say "not transposed"(like `s.transposed()`) instead of strides in descending order, which isn't clear.

Reply:
The tests I made for `find_permutation` does make it look like it gives a permutation that puts the strides in a "not transposed" order.

Reply:
This is the permutation needed to put the `lens` in the same order as they are in memory. So if we start with `NCHW`, each letter could be `0,1,2,3` so then `NHWC` would be `0,2,3,1`. So applying a transpose with the permutation would put the shape in the same order as it is in memory. 

For example, with a NHWC tensor like `{1, 3, 16, 16}, {768, 1, 48, 3}`. The dimensions are in NCHW order, but in memory they are in NHWC order. So `find_permutation` will return `0,2,3,1` and applying a transpose(or `reorder_shape`) with this permutation will yield `{1, 16, 16, 3}, {768, 48, 3, 1}`.